packages/docutils/lib/builder/nav.ts

```diff
@@ -0,0 +1,320 @@
+ */
+export async function updateNav({
+  cwd = process.cwd(),
+  mkdocsYml: mkDocsYmlPath,
+  typedocJson: typeDocJsonPath,
+  all = false,
+  dryRun = false,
+}: UpdateNavOpts = {}) {
+  // we need `mkdocs.yml` to update
+  // and we need `typedoc.json` to know where to look for the command docs
+  [mkDocsYmlPath, typeDocJsonPath] = await Promise.all([
+    mkDocsYmlPath ?? findMkDocsYml(cwd),
+    typeDocJsonPath ?? findTypeDocJsonPath(cwd),
+  ]);
+  if (!mkDocsYmlPath) {
+    throw new DocutilsError(
+      `Could not find ${NAME_MKDOCS_YML} from ${cwd};  run "${NAME_BIN} init" to create it`
+    );
+  }
+  if (!typeDocJsonPath) {
+    throw new DocutilsError(
+      `Could not find ${NAME_TYPEDOC_JSON} from ${cwd};  run "${NAME_BIN} init" to create it`
+    );
+  }
+  const relativePath = relative(cwd);
+  const relMkDocsYmlPath = relativePath(mkDocsYmlPath);
+  const typeDocJson = readTypedocJson(typeDocJsonPath);
+  const mkDocsYml = (await readYaml(mkDocsYmlPath)) as MkDocsYml;
+
+  /**
+   * Absolute path to `typedoc.json`
+   */
+  const absTypeDocJsonPath = path.isAbsolute(typeDocJsonPath)
+    ? typeDocJsonPath
+    : path.resolve(cwd, typeDocJsonPath);
+
+  /**
+   * Absolute path to TypeDoc's output directory (`out`)
+   */
+  const typeDocOutDir = path.resolve(
+    path.dirname(absTypeDocJsonPath),
+    typeDocJson.out ? typeDocJson.out : DEFAULT_REL_TYPEDOC_OUT_PATH
+  );
+
+  /**
+   * Absolute path to `mkdocs.yml`
+   */
+  const absMkdocsYmlPath = path.isAbsolute(mkDocsYmlPath)
+    ? mkDocsYmlPath
+    : path.resolve(cwd, mkDocsYmlPath);
+
+  const {docs_dir: docsDir, nav = []} = mkDocsYml;
+  /**
+   * Absolute path to the directory containing MkDocs input docs
+   */
+  const mkDocsDocsDir = path.resolve(path.dirname(absMkdocsYmlPath), docsDir ?? 'docs');
+
+  /**
+   * @todo
+   * `commands` is a dirname configurable via the `commandsDir` option added by
+   * `@appium/typedoc-plugin-appium`. this lives in `typedoc.json`, but in order for it to be parsed
+   * using TypeDoc's facilities, we have to load plugins before reading `typedoc.json`, which is
+   * slow. we will probably have to support this in the future, but for now, we can just hardcode it
+   */
+  const dirs = all ? await findDirsIn(typeDocOutDir) : [path.join(typeDocOutDir, 'commands')];
+
+  let shouldWriteMkDocsYml = false;
+
+  const navData = parseNav(nav);
+
+  // this is the thing which will be assigned to the `nav` prop
+  // and thus written to `mkdocs.yml` if there were any changes.
+  // we don't need the `name` prop, since the name is already present in the keypath.
+  const newData: Omit<ParsedNavData, 'name'>[] = [];
+
+  for (const dir of dirs) {
+    const newDataForDir: Omit<ParsedNavData, 'name'>[] = [];
+    const newRefFilepaths = await findRelativeMarkdownFiles(dir, mkDocsDocsDir);
+
+    const header = all ? _.startCase(path.basename(dir)) : DEFAULT_NAV_HEADER;
+
+    const headerItems = itemsForHeader(navData, header);
```

Comment:
I usually use verbs for method or function names

Reply:
okay, that's probably better, I will change it